cmake/modules/FindSDL2.cmake

```diff
@@ -68,110 +28,91 @@
+	get_target_property(SDL2_INCLUDE_DIR SDL2::SDL2 INTERFACE_INCLUDE_DIRECTORIES)
+else()
+	set(SDL2_SEARCH_PATHS
+		~/Library/Frameworks
+		/Library/Frameworks
+		/usr/local
+		/usr
+		/sw # Fink
+		/opt/local # DarwinPorts
+		/opt/csw # Blastwave
+		/opt
+		${SDL2_PATH}
 	)
 
-	# For OS X, SDL2 uses Cocoa as a backend so it must link to Cocoa.
-	IF(APPLE)
-		SET_PROPERTY(TARGET SDL2::SDL2 APPEND PROPERTY
-			INTERFACE_LINK_OPTIONS "-framework Cocoa"
-		)
-	ENDIF(APPLE)
-
-	# For threads, as mentioned Apple doesn't need this.
-	# In fact, there seems to be a problem if I used the Threads package
-	# and try using this line, so I'm just skipping it entirely for OS X.
-	IF(NOT APPLE AND Threads_FOUND)
-		SET_PROPERTY(TARGET SDL2::SDL2 APPEND PROPERTY
-			INTERFACE_LINK_LIBRARIES "Threads::Threads"
-		)
-	ENDIF(NOT APPLE AND Threads_FOUND)
-ENDIF(SDL2_LIBRARY)
-
-IF(NOT SDL2_BUILDING_LIBRARY AND SDL2MAIN_LIBRARY)
-	ADD_LIBRARY(SDL2::SDL2main STATIC IMPORTED)
-	SET_TARGET_PROPERTIES(SDL2::SDL2main PROPERTIES
-		IMPORTED_LOCATION "${SDL2MAIN_LIBRARY}"
-		INTERFACE_INCLUDE_DIRECTORIES "${SDL2_INCLUDE_DIR}"
+	find_path(SDL2_INCLUDE_DIR
+		NAMES SDL.h
+		HINTS $ENV{SDL2DIR} ${SDL2_INCLUDE_DIRS}
+		PATH_SUFFIXES SDL2 include/SDL2 include
+		PATHS ${SDL2_SEARCH_PATHS}
 	)
-	IF(MINGW)
-		SET_PROPERTY(TARGET SDL2::SDL2main APPEND PROPERTY
-			INTERFACE_LINK_OPTIONS "${MINGW32_LIBRARY}"
-		)
-	ENDIF(MINGW)
-ENDIF(NOT SDL2_BUILDING_LIBRARY AND SDL2MAIN_LIBRARY)
 
-# message("</FindSDL2.cmake>")
+	if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+		set(PATH_SUFFIXES lib64 lib/x64 lib)
+	else()
+		set(PATH_SUFFIXES lib/x86 lib)
+	endif()
+
+	find_library(SDL2_LIBRARY
+		NAMES SDL2
+		HINTS $ENV{SDL2DIR} ${SDL2_LIBDIR}
+		PATH_SUFFIXES ${PATH_SUFFIXES}
+		PATHS ${SDL2_SEARCH_PATHS}
+	)
 
-INCLUDE(FindPackageHandleStandardArgs)
+	# SDL2 may require threads on your system.
+	# The Apple build may not need an explicit flag because one of the
+	# frameworks may already provide it.
+	# But for non-OSX systems, I will use the CMake Threads package.
+	if(NOT APPLE)
+		find_package(Threads)
+	endif()
+
+	if(SDL2_LIBRARY AND SDL2_INCLUDE_DIR)
+		add_library(SDL2::SDL2 UNKNOWN IMPORTED)
+		set_target_properties(SDL2::SDL2 PROPERTIES
+			IMPORTED_LOCATION "${SDL2_LIBRARY}"
+			INTERFACE_INCLUDE_DIRECTORIES "${SDL2_INCLUDE_DIR}"
+		)
 
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(SDL2 REQUIRED_VARS SDL2_LIBRARY SDL2_INCLUDE_DIR)
+		# For OS X, SDL2 uses Cocoa as a backend so it must link to Cocoa.
+		if(APPLE)
+			set_property(TARGET SDL2::SDL2 APPEND PROPERTY
+				INTERFACE_LINK_OPTIONS "-framework Cocoa"
+			)
+		endif()
+
+		# For threads, as mentioned Apple doesn't need this.
+		# In fact, there seems to be a problem if I used the Threads package
+		# and try using this line, so I'm just skipping it entirely for OS X.
+		if(NOT APPLE AND Threads_FOUND)
+			set_property(TARGET SDL2::SDL2 APPEND PROPERTY
+				INTERFACE_LINK_LIBRARIES "Threads::Threads"
+			)
+		endif()
```

Comment:
This kind of stuff is skipped when SDL2 is found in `CONFIG` mode. Is it safe?

Reply:
One would hope that the config script, since it's provided by SDL2 itself, would add whatever is necessary to use SDL2 on the current system. If not, I would consider that an upstream bug. Also, this was already skipped if SDL2 was found in config mode, since the root `CMakeLists.txt` didn't use module mode if config mode succeeded. The logic is the same, except it's all in the module now.

Reply:
> One would hope that the config script, since it's provided by SDL2 itself, would add whatever is necessary to use SDL2 on the current system.

I checked `INTERFACE_LINK_LIBRARIES` of `SDL2::SDL2` in various build environment. Other than the SDL2 library itself, the property contains:
- Linux: nothing
- macOS(via Homebrew): nothing
- MinGW(via PPA): `-lmingw32 -lSDL2main -mwindows`

I'm not sure if every libraries in this CMake module are required, but I'm pretty sure this CMake module is doing some different from the upstream config script. Not sure which one(maybe both?) is doing wrong...

> Also, this was already skipped if SDL2 was found in config mode, since the root `CMakeLists.txt` didn't use module mode if config mode succeeded.

Alright, it's at least not a regression then.